lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3652,6 +3652,9 @@ ResolvedType GenericSignatureBuilder::maybeResolveEquivalenceClass(
                                    resolutionKind,
                                    wantExactPotentialArchetype);
     if (!resolvedBase) return resolvedBase;
+    if (resolvedBase.getType().is<Type>())
```

Comment:
It's evaluated before the below condition so, yes.

Reply:
I mean, I’m surprised that this doesn’t break code because of that.

Reply:
I'm also a little surprised this didn't break anything. It should be noted that it's checking the base type, not `type`. Then, when the condition succeeds, it will fail out to `handleUnresolvedRequirement`.

Reply:
Is the Type here Null or have an ErrorType in it?

Reply:
As the bots showed, this does cause issues. I'll look into it in more depth in the morning.